Data/Graph.hs

```diff
@@ -23,53 +23,74 @@
 -- Maintainer  :  libraries@haskell.org
 -- Portability :  portable
 --
--- A version of the graph algorithms described in:
+-- = Finite Graphs and Strongly Connected Components
 --
---   /Structuring Depth-First Search Algorithms in Haskell/,
---   by David King and John Launchbury.
+-- The @'Graph'@ type is an adjacency list representation of a finite, directed
+-- graph with @Int@ vertices.
+--
+-- The @'SCC'@ type represents a strongly-connected component of a graph.
+--
+-- == Implementation
+--
+-- The implementation is based on
+--
+--   * /Structuring Depth-First Search Algorithms in Haskell/,
+--     by David King and John Launchbury.
 --
 -----------------------------------------------------------------------------
 
-module Data.Graph(
-
-        -- * External interface
+module Data.Graph (
 
-        -- At present the only one with a "nice" external interface
-        stronglyConnComp, stronglyConnCompR, SCC(..), flattenSCC, flattenSCCs,
+    -- * Graphs
+      Graph
+    , Table
```

Comment:
Side note: I've been wondering whether the `Table` type synonym is doing more harm than good. What happens to signatures if we stop using it (but export it for backwards compatibility)?

Reply:
Personally I don't think its particularly helpful, although maybe adding more comments on it would probably be an improvement.

Reply:
Much better after removing the `Table` type alias from the types.